### ISSUE TITLE
Placeholder affects image scaling when we use CrossfadeTransition and the placeholder image is larger

### ISSUE THREAD

**Issue by tfcporciuncula** [coil-kt/coil#1688]
**Describe the bug**
When I have `CrossfadeTransition.Factory()` passed to the `ImageLoader`, I get a different final scale in an `AsyncImage` if it uses a placeholder that is larger than the final image. Everything is fine if I remove the crossfade, remove the placeholder, or use a placeholder that matches the size of the final image.

**To Reproduce**
Minimum reprodicible example here: https://github.com/tfcporciuncula/coil-crossfade

You can clone the repo, run the code, and observe that:

- If we keep `CrossfadeTransition.Factory()` around, we'll get different scaling results depending on whether we keep or remove the placeholder.
- If we remove `CrossfadeTransition.Factory()`, the scaling is the same regardless if we have the placeholder or not.
- If we keep `CrossfadeTransition.Factory()` but change the placeholder from `R.drawable.placeholder_larger` to `R.drawable.placeholder`, which has the exact same size as our final imagem, then everything is fine.

**Logs/Screenshots**

Image on the left is what we get with crossfade and the large placeholder. Image on the right is the correct scaling which we get by either removing the crossfade, removing the placeholder, or using a placeholder that matches the final image size.

![1](https://user-images.githubusercontent.com/750114/227771995-33a69506-1ba2-4bd2-816a-c4869202d995.png) ![2](https://user-images.githubusercontent.com/750114/227771998-61ddd330-742a-4102-b844-a8c8314d2ce5.png)

We're using `ContentScale.Crop` above, but we can move to `ContentScale.Fit` to make the difference more visible:

![w](https://user-images.githubusercontent.com/750114/227772144-55858b3d-d08d-430a-8559-8fb36b889c41.png) ![c](https://user-images.githubusercontent.com/750114/227772141-8c97fb71-ae7e-48ab-8e38-b2d4d7059c6c.png)

**Version**
Coil 2.3.0, reproduced on a Pixel 4 running Android 13 and on a Pixel 6 emulator (1080x2400 420dpi) running Android 10 (screenshots from the latter).

**Comment by colinrtwhite** [comment]
Thanks for the repro project! At first glance, seems like this could be related to: https://github.com/coil-kt/coil/issues/1505

**Comment by tfcporciuncula** [comment]
True! After reading through it I thought the issue would also happen with a smaller placeholder that has a different aspect ratio, but it doesn’t. The placeholder needs to have a different aspect ratio, but must be larger. I tested it with a larger placeholder that has the same aspect ratio and the issue didn’t happen.

I pushed one more commit in my project adding more placeholder variations and updating my comment just so it’s also easy to verify that:

- Issue doesn’t happen if placeholder is smaller, even if it has a different aspect ratio
- Issue doesn’t happen if the placeholder is larger but has the same aspect ration

**Comment by alexej520** [comment]
Hi!
I forked @tfcporciuncula example, and added working workaround [commit](https://github.com/alexej520/coil-crossfade-issue-1688/commit/b16a9cbbf7470f201d4686bf4bc39028ea6010ac)
The solution is simple: use a copy of CrossfadeTransition instead of original one.
This is a strange solution, but it works for me
